/*
 * IntHelp.java
 *
 * Created on 28 May 2014, 15.42
 */
package org.cytoscape.interference.internal;

/**
 *
 * @author faizaan.shaik
 */
public class IntHelp extends javax.swing.JFrame {

    private String helpString;

    /**
     * Creates new form IntHelp
     */
    public IntHelp() {
        initComponents();
        this.setSize(620, 620);
    }

    /**
     * This method is called from within the constructor to initialize the form.
     * WARNING: Do NOT modify this code. The content of this method is always
     * regenerated by the Form Editor.
     */
    // <editor-fold defaultstate="collapsed" desc="Generated Code">//GEN-BEGIN:initComponents
    private void initComponents() {
        jScrollPane1 = new javax.swing.JScrollPane();
        jTextArea1 = new javax.swing.JTextArea();
        setCursor(new java.awt.Cursor(java.awt.Cursor.DEFAULT_CURSOR));
        setForeground(new java.awt.Color(255, 255, 51));
        jTextArea1.setColumns(20);
        jTextArea1.setRows(5);
        jScrollPane1.setViewportView(jTextArea1);
        org.jdesktop.layout.GroupLayout layout = new org.jdesktop.layout.GroupLayout(getContentPane());
        getContentPane().setLayout(layout);
        layout.setHorizontalGroup(
                layout.createParallelGroup(org.jdesktop.layout.GroupLayout.LEADING).add(layout.createSequentialGroup().addContainerGap().add(jScrollPane1, org.jdesktop.layout.GroupLayout.DEFAULT_SIZE, 518, Short.MAX_VALUE).addContainerGap()));
        layout.setVerticalGroup(
                layout.createParallelGroup(org.jdesktop.layout.GroupLayout.LEADING).add(layout.createSequentialGroup().addContainerGap().add(jScrollPane1, org.jdesktop.layout.GroupLayout.DEFAULT_SIZE, 520, Short.MAX_VALUE).addContainerGap()));
        pack();
    }// </editor-fold>//GEN-END:initComponents

    /**
     * @param args the command line arguments
     */
    public static void main(String args[]) {
        java.awt.EventQueue.invokeLater(new Runnable() {

            public void run() {
                new IntHelp().setVisible(true);
            }
        });
    }
    // Variables declaration - do not modify//GEN-BEGIN:variables
    private javax.swing.JScrollPane jScrollPane1;
    private javax.swing.JTextArea jTextArea1;
    // End of variables declaration//GEN-END:variables

    public void setText(int centrality) {
        switch (centrality) {
            case 0: {
                helpString
                        = "DIAMETER: description  \n\n"
                        + "The diameter is the maximal distance (shortest path) amongst all the distances"
                        + "calculated between each couple of vertexes in graph."
                        + "The diameter indicates how much distant are the two most distant nodes. It"
                        + "can be a first and simple general parameter of graph \"compactness\", meaning"
                        + "with that the overall proximity between nodes. A \"high\" graph diameter "
                        + "indicates that the two nodes determining that diameter are very distant, "
                        + "implying little graph compactness. However, it is possible that two nodes "
                        + "are very distant, thus giving a high graph diameter, but several other nodes "
                        + "are not. Therefore, a graph could have high diameter and still being rather "
                        + "compact or have very compact regions. Thus, a high graph diameter can be "
                        + "misleading in term of evaluation of graph compactness. In contrast a \"low\"  "
                        + "graph diameter is much more informative and reliable. Indeed, a low diameter "
                        + "surely indicates that all the nodes are in proximity and the graph is "
                        + "compact. In quantitative terms, \"high\" and \"low\" are better defined when "
                        + "compared to the total number of nodes in the graph. Thus, a low diameter of "
                        + "a very big graph (with hundreds of nodes) is much more meaningful in term of "
                        + "compactness than a low diameter of a small graph (with few nodes). Notably, "
                        + "the diameter enables to measure the development of a network in time.  \n\n"
                        + "IN BIOLOGICAL TERMS\n\nThe diameter, and thus the compactness, of a biological "
                        + "network, for instance a protein-signaling network, can be interpreted as the "
                        + "overall easiness of the proteins to communicate and/or influence their "
                        + "reciprocal function. It could be also a sign of functional convergence. "
                        + "Indeed, a big protein network with low diameter may suggest that the "
                        + "proteins within the network had a functional co-evolution. The diameter "
                        + "should be carefully weighted if the graph is not fully connected (that is, "
                        + "there are isolated nodes). ";
                this.setTitle("Interference Help: Diameter ");
            }
            break;
            case 1: {
                helpString
                        = "AVERAGE DISTANCE: description  \n\n"
                        + "The average distance is the average shortest path of a graph, corresponding  "
                        + "to the summa of all shortest paths between vertex couples divided for the  "
                        + "total number of vertex couples. Often it is not an integer.  "
                        + "As for the diameter, it can be a simple and general parameter of graph  "
                        + "\"compactness\", meaning with that the overall tendency of nodes to stay in  "
                        + "proximity. Being an average, it can be somehow more informative than the  "
                        + "diameter and can be also considered a general indicator of network  "
                        + "\"navigability\". A \"high\" average distance indicates that the nodes are  "
                        + "distant (disperse), implying little graph compactness. In contrast a \"low\"  "
                        + "average distance indicates that all the nodes are in proximity and the graph  "
                        + "is compact. In quantitative terms, \"high\" and \"low\" are better defined when  "
                        + "compared to the total number of nodes in the graph. Thus, a low average  "
                        + "distance of a very big graph (with hundreds of nodes) is more meaningful in  "
                        + "term of compactness than a low average distance of a small graph (with few  "
                        + "nodes).  \n\n"
                        + "IN BIOLOGICAL TERMS\n\nThe average distance of a biological network, for  "
                        + "instance a protein-signaling network, can be interpreted as the overall  "
                        + "easiness of the proteins to communicate and/or influence their reciprocal  "
                        + "function. It could be also a sign of functional convergence. Indeed, a big  "
                        + "protein network with low average distance may suggest that the proteins  "
                        + "within the network have the tendency to generate functional complexes and/or  "
                        + "modules (although centrality indexes should be also calculated to support  "
                        + "that indication).";

                this.setTitle("Interference Help: Average distance ");
            }
            break;

            case 2: {
                helpString
                        = "DEGREE: description   \n\n"
                        + "The degree is the simplest topological index, corresponding to the number of nodes  "
                        + "adjacent to a given node v, where \"adjacent\" means directly connected. The  "
                        + "nodes directly connected to a given node v are also called \"first neighbors\"  "
                        + "of the given node. Thus, the degree also corresponds to the number of  "
                        + "adjacent incident edges. In directed networks we distinguish in-degree, when  "
                        + "the edges target the node v, and out-degree, when the edges target the  "
                        + "adjacent neighbors of v.  \n\n"
                        + "IN BIOLOGICAL TERMS\n\nThe degree allows an immediate evaluation of the  "
                        + "regulatory relevance of the node. For instance, in signaling networks,  "
                        + "proteins with very high degree are interacting with several other signaling  "
                        + "proteins, thus suggesting a central regulatory role, that is they are likely  "
                        + "to be regulatory \"hubs\". For instance, signaling proteins encoded by  "
                        + "oncogenes, such as HRAS, SRC or TP53, are hubs. Depending on the nature of  "
                        + "the protein, the degree could indicate a central role in amplification  "
                        + "(kinases), diversification and turnover (small GTPases), signaling module  "
                        + "assembly (docking proteins), gene expression (transcription factors), etc.  "
                        + "Signaling networks have typically a scale-free architecture.  ";

                this.setTitle("Interference Help: Node degree ");
            }
            break;

            case 3: {
                helpString
                        = "ECCENTRICITY: description  \n\n"
                        + "The eccentricity is a node centrality index. The eccentricity of a node v is  "
                        + "calculated by computing the shortest path between the node v and all other  "
                        + "nodes in the graph, then the \"longest\" shortest path is chosen (let (v, K)  "
                        + "where K is the most distance node from v). Once this path with length  "
                        + "dist(v, K) is identified, its reciprocal is calculated (1/dist(v, K)). By  "
                        + "doing that, an eccentricity with higher value assumes a positive meaning in  "
                        + "term of node proximity. Indeed, if the eccentricity of the node v is high,  "
                        + "this means that all other nodes are in proximity. In contrast, if the  "
                        + "eccentricity is low, this means that there is at least one node (and all its  "
                        + "neighbors) that is far form node v. Of course, this does not exclude that  "
                        + "several other nodes are much closer to node v. Thus, eccentricity is a more  "
                        + "meaningful parameter if is high. Notably, \"high\" and \"low\" values are more  "
                        + "significant when compared to the average eccentricity of the graph G  "
                        + "calculated by averaging the eccentricity values of all nodes in the graph.  \n\n"
                        + "IN BIOLOGICAL TERMS\n\nThe eccentricity of a node in a biological network, for  "
                        + "instance a protein-signaling network, can be interpreted as the easiness of  "
                        + "a protein to be functionally reached by all other proteins in the network.  "
                        + "Thus, a protein with high eccentricity, compared to the average eccentricity  "
                        + "of the network, will be more easily influenced by the activity of other  "
                        + "proteins (the protein is subject to a more stringent or complex regulation)  "
                        + "or, conversely could easily influence several other proteins. In contrast, a  "
                        + "low eccentricity, compared to the average eccentricity of the network, could  "
                        + "indicate a marginal functional role (although this should be also evaluated  "
                        + "with other parameters and contextualized to the network annotations).  ";
                this.setTitle("Interference Help: Eccentricity ");
            }
            break;

            case 4: {
                helpString
                        = "RADIALITY: description  \n\n"
                        + "The radiality is a node centrality index. The radiality of a node v is  "
                        + "calculated by computing the shortest path between the node v and all other  "
                        + "nodes in the graph. The value of each path is then subtracted by the value  "
                        + "of the diameter + 1 and the resulting values are summated. Finally,  "
                        + "the obtained value is divided for the number of nodes -1 (n-1). Basically,  "
                        + "as the diameter is the maximal possible distance between nodes, subtracting  "
                        + "systematically from the diameter the shortest paths between the node v and  "
                        + "its neighbors will give high values if the paths are short and low values if  "
                        + "the paths are long. Overall, if the radiality is high this means that, with  "
                        + "respect to the diameter, the node is generally closer to the other nodes,  "
                        + "whereas, if the radiality is low, this means that the node is peripheral.  "
                        + "Also here, \"high\" and \"low\" values are more meaningful when compared to the  "
                        + "average radiality of the graph G calculated by averaging the radiality  "
                        + "values of all nodes in the graph.  "
                        + "As for the closeness, the radiality value should be considered as an  "
                        + "\"average tendency to node proximity or isolation\", not definitively  "
                        + "informative on the centrality of the individual node. The radiality should  "
                        + "be always compared to the closeness and to the eccentricity: a node with  "
                        + "high eccentricity + high closeness+ high radiality is a consistent  "
                        + "indication of a high central position in the graph.  \n\n"
                        + "IN BIOLOGICAL TERMS\n\nThe radiality of a node in a biological network, for  "
                        + "instance a protein-signaling network, can be interpreted as the  "
                        + "\"probability\" of a protein to be functionally relevant for several other  "
                        + "proteins, but with the possibility to be irrelevant for few other proteins.  "
                        + "Thus, a protein with high radiality, compared to the average radiality of  "
                        + "the network, will be easily central to the regulation of other proteins but  "
                        + "with some proteins not influenced by its activity.  Notably, in biological  "
                        + "networks could be also of interest to analyze proteins with low radiality,  "
                        + "compared to the average radiality of the network, as these proteins,  "
                        + "although less relevant for that specific network, are possibly behaving as  "
                        + "intersecting boundaries with other networks. Accordingly, a signaling  "
                        + "network with a very high average radiality is more likely organizing  "
                        + "functional units or modules, whereas a signaling network with very low  "
                        + "average radiality will behave more likely as an open cluster of proteins  "
                        + "connecting different regulatory modules. All these interpretations should be  "
                        + "accompanied to the contemporary evaluation of eccentricity and closeness.  ";

                this.setTitle("Interference Help: Radiality");
            }
            break;

            case 5: {
                helpString
                        = "CLOSENESS: description  \n\n"
                        + "The closeness is a node centrality index. The closeness of a node v is  "
                        + "calculated by computing the shortest path between the node v and all other  "
                        + "nodes in the graph, and then calculating the summa. Once this value is  "
                        + "obtained, its reciprocal is calculated, so higher values assume a positive  "
                        + "meaning in term of node proximity. Also here, \"high\" and \"low\" values are  "
                        + "more meaningful when compared to the average closeness of the graph G  "
                        + "calculated by averaging the closeness values of all nodes in the graph.  "
                        + "Notably, high values of closeness should indicate that all other nodes are  "
                        + "in proximity to node v. In contrast, low values of closeness should indicate  "
                        + "that all other nodes are distant from node v. However, a high closeness  "
                        + "value can be determined by the presence of few nodes very close to node v,  "
                        + "with other much more distant, or by the fact that all nodes are generally  "
                        + "very close to v. Likewise, a low closeness value can be determined by the  "
                        + "presence of few nodes very distant from node v, with other much closer, or  "
                        + "by the fact that all nodes are generally distant from v. Thus, the closeness  "
                        + "value should be considered as an \"average tendency to node proximity or  "
                        + "isolation\", not really informative on the specific nature of the individual  "
                        + "node couples. The closeness should be always compared to the eccentricity: a  "
                        + "node with high eccentricity + high closeness is very likely to be central in  "
                        + "the graph.  \n\n"
                        + "IN BIOLOGICAL TERMS\n\nThe closeness of a node in a biological network, for  "
                        + "instance a protein-signaling network, can be interpreted as the  "
                        + "\"probability\" of a protein to be functionally relevant for several other  "
                        + "proteins, but with the possibility to be irrelevant for few other proteins.  "
                        + "Thus, a protein with high closeness, compared to the average closeness of  "
                        + "the network, will be easily central to the regulation of other proteins but  "
                        + "with some proteins not influenced by its activity.  Notably, in biological  "
                        + "networks could be also of interest to analyze proteins with low closeness,  "
                        + "compared to the average closeness of the network, as these proteins,  "
                        + "although less relevant for that specific network, are possibly behaving as  "
                        + "intersecting boundaries with other networks. Accordingly, a signaling  "
                        + "network with a very high average closeness is more likely organizing  "
                        + "functional units or modules, whereas a signaling network with very low  "
                        + "average closeness will behave more likely as an open cluster of proteins  "
                        + "connecting different regulatory modules.  ";

                this.setTitle("Interference Help: Closeness");
            }
            break;

            case 6: {
                helpString
                        = "STRESS: description  \n\n"
                        + "The stress is a node centrality index. Stress is calculated by measuring the  "
                        + "number of shortest paths passing through a node. To calculate the \"stress\"  "
                        + "of a node v, all shortest paths in a graph G are calculated and then the  "
                        + "number of shortest paths passing through v is counted. A \"stressed\" node is  "
                        + "a node traversed by a high number of shortest paths. Notably, and  "
                        + "importantly, a high stress values does not automatically implies that the  "
                        + "node v is critical to maintain the connection between nodes whose paths are  "
                        + "passing through it. Indeed, it is possible that two nodes are connected by  "
                        + "means of other shortest paths not passing through the node v. Also here,  "
                        + "\"high\" and \"low\" values are more meaningful when compared to the average  "
                        + "stress value of the graph G calculated by averaging the stress values of all  "
                        + "nodes in the graph.  \n\n"
                        + "IN BIOLOGICAL TERMS\n\nThe stress of a node in a biological network, for  "
                        + "instance a protein-signaling network, can indicate the relevance of a  "
                        + "protein as functionally capable of holding together communicating nodes. The  "
                        + "higher the value the higher the relevance of the protein in connecting  "
                        + "regulatory molecules. Due to the nature of this centrality, it is possible  "
                        + "that the stress simply indicates a molecule heavily involved in cellular  "
                        + "processes but not relevant to maintain the communication between other  "
                        + "proteins.";

                this.setTitle("Interference Help: Stress");
            }
            break;

            case 7: {
                helpString
                        = "BETWEENNESS: description  \n\n"
                        + "The S.-P. Betweenness is a node centrality index. It is similar to the  "
                        + "stress but provides a more elaborated and informative centrality index. It  "
                        + "is calculated considering couples of nodes (v1, v2) and counting the number  "
                        + "of shortest paths linking v1 and v2 and passing through a node n. Then, the  "
                        + "value is related to the total number of shortest paths linking v1 and v2.  "
                        + "Thus, a node can be traversed by only one path linking v1 and v2, but if  "
                        + "this path is the only connecting v1 and v2 the node n will score a higher  "
                        + "betweenness value (in the stress computation would have had a low score).  "
                        + "Thus, a high S.-P. Betweenness score means that the node, for certain paths,  "
                        + "is crucial to maintain node connections. Notably, to know the number of  "
                        + "paths for which the node is critical it is necessary to look at the stress.  "
                        + "Thus, stress and S.-P. Betweenness can be used to gain complementary  "
                        + "information. Further information could be gained by referring the S.-P.  "
                        + "Betweenness to node couples, thus \"quantifying\" the importance of a node for  "
                        + "two connected nodes. Also here, \"high\" and \"low\" values are more meaningful  "
                        + "when compared to the average S.-P. Betweenness value of the graph G  "
                        + "calculated by averaging the S.-P. Betweenness values of all nodes in the  "
                        + "graph.  \n\n"
                        + "IN BIOLOGICAL TERMS\n\nThe S.-P. Betweenness of a node in a biological  "
                        + "network, for instance a protein-signaling network, can indicate the  "
                        + "relevance of a protein as functionally capable of holding together  "
                        + "communicating proteins. The higher the value the higher the relevance of the  "
                        + "protein as organizing regulatory molecule. The S.-P. Betweenness of a  "
                        + "protein effectively indicates the capability of a protein to bring in  "
                        + "communication distant proteins. In signaling modules, proteins with high  "
                        + "S.-P. Betweenness are likely crucial to maintain functionally and coherence  "
                        + "of signaling mechanisms.";
                this.setTitle("Interference Help: Betweenness");
            }
            break;

            case 8: {
                helpString
                        = "CENTROID VALUE: description  \n\n"
                        + "The centroid value is the most complex node centrality index. It is computed  "
                        + "by focusing the calculus on couples of nodes (v, w) and systematically  "
                        + "counting the nodes that are closer (in term of shortest path) to v or to w.  "
                        + "The calculus proceeds by comparing the node distance from other nodes with  "
                        + "the distance of all other nodes from the others, such that a high centroid  "
                        + "value indicates that a node v is much closer to other nodes. Thus, the  "
                        + "centroid value provides a centrality index always weighted with the values  "
                        + "of all other nodes in the graph. Indeed, the node with the highest centroid  "
                        + "value is also the node with the highest number of neighbors (not only first)  "
                        + "if compared with all other nodes. In other terms, a node v with the highest  "
                        + "centroid value is the node with the highest number of neighbors separated by  "
                        + "the shortest path to v. The centroid value suggests that a specific node has  "
                        + "a central position within a graph region characterized by a high density of  "
                        + "interacting nodes. Also here, \"high\" and \"low\" values are more meaningful  "
                        + "when compared to the average centrality value of the graph G calculated by  "
                        + "averaging the centrality values of all nodes in the graph.  \n\n"
                        + "IN BIOLOGICAL TERMS\n\nThe centrality value of a node in a biological network,  "
                        + "for instance a protein-signaling network, can be interpreted as the  "
                        + "\"probability\" of a protein to be functionally capable of organizing discrete  "
                        + "protein clusters or modules. Thus, a protein with high centroid value,  "
                        + "compared to the average centroid value of the network, will be possibly  "
                        + "involved in coordinating the activity of other highly connected proteins,  "
                        + "altogether devoted to the regulation of a specific cell activity (for  "
                        + "instance, cell adhesion, gene expression, proliferation etc.). Accordingly,  "
                        + "a signaling network with a very high average centroid value is more likely  "
                        + "organizing functional units or modules, whereas a signaling network with  "
                        + "very low average centroid value will behave more likely as an open cluster  "
                        + "of proteins connecting different regulatory modules. It can be useful to  "
                        + "compare the centroid value to other algorithms detecting dense regions in a  "
                        + "graph, indicating protein clusters, such as, for instance, MCODE.";

                this.setTitle("Interference Help: Centroid value");
            }
            break;
            case 9: {
                helpString
                        = "EIGENVECTOR CENTRALITY: Description\n\n"
                        + "Eigenvector centrality is a node centrality index. It assigns \n"
                        + "relative scores to all nodes in a network based on the concept that\n"
                        + "connections to high-scoring nodes contribute to the score of a specific\n"
                        + "node more than equal connections to low-scoring nodes. The definition is\n"
                        + "recursive: an high Eigenvector value means that a node has many neighbors\n"
                        + "with high Eigenvector value. So, the Eigenvector centrality can be viewed\n"
                        + "as a sort of weighted degree, where not only the number of the neighbors is\n"
                        + "important but also the score (the Eigenvector itself) of the neighbors. A high\n"
                        + "Eigenvector centrality indicates that a node is being visited a lot while\n"
                        + "traversing and is well-connected. A variant of the Eigenvector algorithm is\n"
                        + "used by  Google's Page Rank algorithm.\n\n"
                        + "IN BIOLOGICAL TERMS\n\n"
                        + "Eigenvector allows an immediate, rather informative, evaluation of the\n"
                        + "regulatory relevance of the node. For instance, a protein with a very high\n"
                        + "Eigenvector is a protein interacting  with several important proteins (regulating\n"
                        + "them or being regulated by them),  thus suggesting a central super-regulatory\n"
                        + "role or a critical target of a regulatory pathway. In contrast, a protein with \n"
                        + "low Eigenvector, can be considered a peripheral protein, interacting with few\n"
                        + "and not central proteins. In a context of directed networks, calculation of\n"
                        + "Eigenvector could be useful to estimate the output of the pharmacological\n"
                        + "modulation of a specific protein.  Indeed, pharmacological modulation (for\n"
                        + "instance inhibition) of a high Eigenvector protein can generate broad biological\n"
                        + "effects but potentially characterized by consistent side-effects, whereas,\n"
                        + "pharmacological modulation of a low Eigenvector protein could generate limited,\n"
                        + "more focused, effects with minimal side-effects. Thus, depending of the context,\n"
                        + "the Eigenvector could potentially help to identify best treatments for diseases.\n";
                this.setTitle("CentiScaPe Help: EigenVector Centrality");
            }
            break;
            case 10: {
                helpString = "BRIDGING CENTRALITY: Description\n\n"
                        + "The Bridging Centrality of a node is the product of the Bridging Coefficient (BC)\n"
                        + "and the Betweenness centrality (Btw). The Bridging Coefficient of a node is\n"
                        + "high if the node has neighbors with high degree. This means that the bridging\n"
                        + "coefficient is high if the node has highly connected first neighbors. Since the\n"
                        + "Bridging Centrality is equals to BC*Btw, this means that a node has high Bridging\n"
                        + "Centrality if it is a connecting node (due to Btw component) of high degree nodes\n"
                        + "(due to BC component). This can suggest that the node is a connecting node between\n"
                        + "clusters or between dense region of the network.  \n\n"
                        + "IN BIOLOGICAL TERMS\n\n "
                        + "The Btw component of the Bridging centrality, suggests that in a biological network\n"
                        + "such as, for example, a protein-protein interaction network, a  protein with high\n"
                        + "Bridging Centrality value is functionally capable of holding together communicating\n"
                        + "proteins. The BC components may also suggest that such communicating proteins are\n"
                        + "regulatory proteins since they interact with many other proteins. Thus, a protein\n"
                        + "with high Bridging centrality is a protein possibly bringing in communication sets\n"
                        + "of regulatory proteins. In a context of directed networks, a high value of Bridging\n"
                        + "Centrality may indicate that a protein is regulated by many proteins at once or that\n"
                        + "is regulating the overall architecture of many clusters at once.\n";
                this.setTitle("CentiScaPe Help: Bridging Centrality");
            }
            break;
            case 11: {
                helpString = "EDGE BETWEENNESS CENTRALITY: Description\n\n"
                        + "The Edge betweenness centrality is an edge centrality index. It is the edge version\n"
                        + "of the node betweenness centrality. Given an edge E, it is calculated considering\n"
                        + "couples of nodes (v1, v2) and counting the number of shortest paths linking v1 and v2\n"
                        + "and passing through the edge E. Then, the value is related to the total number of\n"
                        + "shortest paths linking v1 and v2. Thus, an edge can be traversed by only\n"
                        + "one path linking v1 and v2, but if this path is the only connecting v1 and v2 the edge\n"
                        + "will score a higher betweenness value. Thus, a high S.-P. Betweenness score means\n"
                        + "that the edge, for certain paths, is crucial to maintain node connections.\n"
                        + "Edge betweenness scores are reported in the edge attribute table.\n\n"
                        + "IN BIOLOGICAL TERMS\n\n"
                        + "Since the Edge-Betweenness is an edge score of centrality, with the edge indicating\n"
                        + "the functional relationship between two nodes (for instance a kinase-triggered\n"
                        + "phosphorylation), in a biological network such as, for instance, a protein-signaling\n"
                        + "network, computation of this centrality moves the focus from node relevance to the\n"
                        + "biochemical process relevance, thus indicating that a specific biochemical reaction\n"
                        + "(or any other type of functional relationship) has a central role in the network\n"
                        + "functional organization. The higher the value the higher the relevance of the\n"
                        + "interaction as organizing regulatory process. In signaling modules, interactions\n"
                        + "with high E-Betweenness are likely prevalent to maintain functionally and coherence\n"
                        + "of signaling mechanisms. In a metabolic network, a high Edge Betweenness value is\n"
                        + "related to the relevance of a reaction in energy and mass processing. Overall,\n"
                        + "Edge Betweenness quantify the capability of a certain reaction  to be a information\n"
                        + "(signaling) or mass (metabolic) \"transit\" reaction, a sort of  obligatory passage\n"
                        + "point for the reactions modeling the network.\n";
                this.setTitle("CentiScaPe Help: Edge Betweenness Centrality");
            }
            break;

            case 12: {
                helpString = "UNDIRECTED NETWORKS\n\n"
                        + "Undirected networks are networks where there is no direction of the edges. The edges\n"
                        + "are considered as bi-directional and the classic centralities definition are applied.\n";

                this.setTitle("CentiScaPe Help: Undirected networks");
            }
            break;
            case 13: {
                helpString = "DIRECTED NETWORKS\n\n"
                        + "Now centralities can be computed also for directed network. A directed network is a\n"
                        + "network where the edge has a direction. This means that given two nodes A and B, some\n"
                        + "times there are no paths betweens them, or there is a path from node A and B but not from\n"
                        + "node B to node A. This is very typical of signaling or metabolic networks, where\n"
                        + "informational or energy/mass flow follows a specific direction determined by the\n"
                        + "thermodynamics of biochemical reactions. Some of the centralities definitions cannot\n"
                        + "be used in directed networks, so they have been redefined in order to consider direction\n"
                        + "of the edges. ";
                this.setTitle("CentiScaPe Help: Directed networks");
            }
            break;
            case 14: {
                helpString = "WEIGHTED NETWORKS\n"
                        + "\n"
                        + "Centralities are based on the shortest path. In a weighted networks, the edges have an\n"
                        + "attribute value that is considered during the computation of the shortest-path. This\n"
                        + "value is used as a \"distance\" between nodes considered in the evaluation of the\n"
                        + "shortest paths between nodes. It means that if you have two paths connecting two\n"
                        + "nodes with the same number of edges but with different weights, only the shortest\n"
                        + "one will be considered, using the provided distance.\n"
                        + "\n"
                        + "\n"
                        + "Suggestion 1: Provide the right values for the edge weight\n"
                        + "\n"
                        + "Example \n"
                        + "\n"
                        + "Suppose you have two paths connecting node A to node D.\n"
                        + "the paths are:\n"
                        + "A -> B -> D\n"
                        + "and \n"
                        + "A -> C -> D\n"
                        + "where the weight are \n"
                        + "w(A-B) = 1\n"
                        + "w(B-D) = 1\n"
                        + "w(A-C) = 2\n"
                        + "w(C-D) = 1\n"
                        + "Here the shortest path connecting A and D is A-B-C and the total length is 2. The length\n"
                        + "of the path A-C-D is 3 and it is not considered, even if the number of edges of the two\n"
                        + "paths is the same.\n"
                        + "\n"
                        + "Take care to consider the weight of the network in a proper way: If the weight means \"space\"\n"
                        + "or \"time\" it can be used, if the weight means \"speed\" you have to use 1/weight as attribute\n"
                        + "of the edge.\n"
                        + "\n"
                        + "\n"
                        + "Suggestion 2: Provide values for the edge weight that are properly rounded\n"
                        + "\n"
                        + "Example\n"
                        + "\n"
                        + "Suppose you have two paths connecting node A to node D. \n"
                        + "the paths are:\n"
                        + "A -> B -> D\n"
                        + "and \n"
                        + "A -> C -> D\n"
                        + "where the weights are \n"
                        + "w(A-B) = 1\n"
                        + "w(B-D) = 1\n"
                        + "w(A-C) = 1,01\n"
                        + "w(C-D) = 1\n"
                        + "\n"
                        + "Here the shortest path connecting A and D is A-B-C and the total length is 2. The length\n"
                        + "of the path A-C-D is 2,01 but it is not considered since the difference is very low. If you\n"
                        + "round the value of distance A-C from 1.01 to 1, the path A-C-D is also considered, since its\n"
                        + "distance become 2 (the same of the other path). \n"
                        + "\n"
                        + "Take care of consider the values properly rounded depending on the meaning of your edge attribute.\n";
                this.setTitle("CentiScaPe Help: Weighted networks");
            }
            break;
            case -1: {
                helpString = "Nodes Options\n\n"
                        + "-If you have selected only one node choose the Single nodes option.\n"
                        + "-If you have selected more than one node, you can have two options:\n\n"
                        + "Single nodes: This remove the nodes one by one.\n\n"
                        + "It means that the first node is removed and then interference is computed. "
                        + "After the computation the node is re-added to the network and the computation restart "
                        + "removing the second node. "
                        + "This process is repeated for all the nodes you selected. You will have one computation and one "
                        + "final result for each selected node. "
                        + "The results are the interference values of all the single nodes"
                        + "with respect to the overall network."
                        + "In a protein network this correspond to inhibit one protein at a time in the virtual experiment.\n\n"
                        + "Set of nodes: This remove all the selected nodes at the same time (as a unique meta-node).\n\n"
                        + "It means that all the nodes are removed from the network and then interference is computed."
                        + " You will have only one computation and only one result (for each centrality) that are the "
                        + "interference values of the set of nodes with respect to the overall network."
                        + " In a protein network this correspond to inhibit all the "
                        + "selected proteins at the same time in the virtual experiment.\n\n";

                this.setTitle("Interference Help: Nodes options");
            }
            break;
        }
        String explaininterference = "INTERFERENCE \n\nInterference evaluates the topological effects of single or multiple nodes removal. "
                + "In this perspective, interference allows virtual node knock-out experiments: "
                + "It is possible to remove one or more nodes "
                + "from the network and analyze the consequences on network structure, "
                + "by looking to the variations of the node centralities values. "
                + "As the centrality value of a node is strictly dependent on the network structure "
                + "and on the properties of other nodes in the network, "
                + "the consequences of a node deletion are well captured by the variation "
                + "on the centrality values of all the other nodes.\n\n"
                + "How Interference plugin works \n\n"
                + "-Computes the centralities value of the network\n"
                + "-Removes the node(s) of interest\n"
                + "-Recompute the centralities in the new network (the one where the  node(s) have been removed)\n"
                + "-Evaluate the differences between the centralities in the two networks.\n\n"
                + "The interference results are expressed as percentage of the total centrality value. "
                + "This allows identifying the differences between the two networks: "
                + "the one with the node(s) of interest and the one where the node(s) have been removed."
                + "Some nodes will increase, whereas others will reduce, "
                + "their individual centrality values.  "
                + "Depending on the sign of the interference value we can have two different cases:   \n\n"
                + "Positive Interference   \n\n"
                + "If a node (A), upon removal from the network of a specific node (B) "
                + "or of a group of nodes, decreases its value for a certain centrality index, "
                + "its interference value is positive. "
                + "This means that this node (A), topologically speaking, "
                + "takes advantage (is positively influenced) by the presence "
                + "in the network of the node (B) or of that group of nodes. "
                + "Thus, removal of node (B) or of that group of nodes "
                + "from the network, negatively affects the topological role of the node (A). "
                + "This is called positive interference.\n\n"
                + "Negative Interference\n\n"
                + "If a node (A), upon removal from the network of a specific node (B) "
                + "or of a group of nodes, increases its value for a certain centrality index, "
                + "its interference value is negative. "
                + "This means that this node (A), topologically speaking, "
                + "is disadvantaged (is negatively influenced) by the presence "
                + "in the network of the node (B) or of that group the nodes. "
                + "Thus, removal of node (B) or of that group of nodes from "
                + "the network, positively affects the topological role of node (A). "
                + "This is called negative interference.\n\n"
                + "The Interference meaning also on the centrality for which it is computed. "
                + "See below: \n\n ";
        if (centrality >= 0 && centrality <= 8) {
            helpString = explaininterference + helpString + "\n\n";
        }
        jTextArea1.setText(helpString);
        jTextArea1.setCaretPosition(0);
    }
}
